### DISCUSSION TITLE
Need help installing benotes on synology

### DISCUSSION THREAD

**Discussion by Sylvio82** [fr0tt/benotes#65] [General]
Hi.
I am not a synology or docker expert, but I already installed some docker containers on my synology.
But I never tried to install a container like benotes.

**1. wget -O .env https://raw.githubusercontent.com/fr0tt/benotes/master/.env.sqlite.example**
This works. I downloaded this file on my windows machine

**2. copy the content of this [file](https://raw.githubusercontent.com/fr0tt/benotes/master/.env.sqlite.example) into a new file called .env**
This works. I copied the content of your file to a file called .env on my machine

**3. edit .env by setting a random secret for APP_KEY (generated with e.g. openssl rand -base64 32) and JWT_SECRET**
??? I dont know what to do here. I have to set a key, but what does the rest mean?

**Comment by fr0tt** [comment]
Hi,

you just have to set a random value for APP_KEY and JWT_SECRET

Feel free to ask again if you should have any more questions

**Reply by Sylvio82** [reply]
I think I have an elementary problem of understanding your guide.

Usually all my docker containers are in /volume1/docker/name_of_the_app. First thing I do is create an app folder in /docker and sometimes some subfolders like data, config... where everything is saved.

So usually my docker script looks like this:

**docker container run -d --restart unless-stopped --name=tvheadend --net=host \
-e PUID=1029 \
-e PGID=100 \
-e TZ=Europe/Berlin \
-v /volume1/docker/tvheadend/recordings:/recordings \
-v /volume1/docker/tvheadend/config:/config \
-v /volume1/docker/tvheadend/picons:/picons \
-v /etc/localtime:/etc/localtime:ro \
linuxserver/tvheadend:amd64-latest**

And now I have no idea how the edit your script to start the container. If I start it without editing I get the error message: 
**docker: open ./volume1/docker/benotes/.env: no such file or directory** (.env file is already copied to volume1/docker/benotes)

**Reply by fr0tt** [reply]
Did you have .../docker/benotes appropriate permissions in order for the container to write and read in it ? 

Maybe this debate might be of help to you as well or the reason why it doesn't work with that image in the moment: https://github.com/fr0tt/benotes/issues/43

**Comment by Sylvio82** [comment]
I think I am too dumb to configure your script the right way :). I dont understand your volume variables. I added the correct uid to have rights for the docker folder.
My script looks like this now, but I get the following error: **/bin/bash: line 3: -v: command not found**

docker run -p 8000:80 -i --rm --name=benotes \
-e USER_ID=1031 \
-e GROUP_ID=100 \    
-v /volume1/docker/benotes/benotes_storage:/var/www/storage \
-v "$(pwd)"/nginx/logs/:/var/lib/nginx/logs/ \
-v "$(pwd)"/.env:/var/www/.env \
--env-file ./volume1/docker/benotes/.env \
--name benotes fr0tt/benotes

**Reply by fr0tt** [reply]
I'm not exactly sure if that's the issue but in line 3 you use /volume1/... where as later you use ./volume1/... 
Also you specify the name twice (in the first and last line).

But the bigger/next issue is that benotes_storage is supposed to be a named volume instead of a bind mount (because storage contains a lot of different subfolders). So you either change that or you really want to use bind mounts instead have a look at [this](https://github.com/fr0tt/benotes/issues/30#issuecomment-1147791893)

**Reply by Bag72** [reply]
I suspect that the reason you are getting the -v error is that you have the docker text in a script or are typing it in directly.    If it is in a script you need a \ at the end of each line or it thinks it is a new command.   so put a space and \ at the end of each line.
pwd gives the current directory so you may as well get rid of the "$(pwd)".

go to the directory you have your docker files in.  Create a new one for benotes, move into it
mkdir benotes
cd benotes
copy your ,env into there
nano benotes.sh
put in
docker run -p 9975:80 -i --rm --name=benotes \
-v ./benotes_storage:/var/www/storage \
-v .env:/var/www/.env \
fr0tt/benotes
Then execute it
sh benotes.sh
When done.
docker logs -f benotes
See what that shows and use <ctrl>c when you want to exit the log output  It should stop on 'NOTICE: ready to handle connections'
Then try connecting to it in your browser I assume it will be on your <NAS IP>     for example http://123.123.123.123:9975

**Comment by Sylvio82** [comment]
Ok. I had some time to test a little more ;)
I tested my docker cli step by step until the first error appears. For now it looks like this:

_docker run -p 5678:5678 -i --rm \
-e USER_ID=1031 \
-e GROUP_ID=100 \
-v /volume1/docker/benotes/benotes_storage:/var/www/storage \
-v "$(pwd)"/nginx/logs/:/var/lib/nginx/logs/ \
--name benotes fr0tt/benotes_

I had to change the port, because I got the error: **Bind for [0.0.0.0:8000](http://0.0.0.0:8000/) failed: port is already allocated**. I also changed this port in the .env file.

I had to remove the t argument because I got the error: **the input device is not a TTY**

The latest error I get is: docker: Error response from daemon: **Bind mount failed: '/root/nginx/logs' does not exists**

What do I have to do here?

**Comment by fr0tt** [comment]
I assume you were in `/root` while executing this? You need to create a directory for this application and inside of it execute the docker run command.

Also your `/volume1/docker/benotes/benotes_storage:/var/www/storage` bind mount might not prevent the container from booting but it won't work properly. Have a look at my previous response.

**Reply by Sylvio82** [reply]
Sorry I cant follow your advice. I am not a docker expert, but I already installed a lot of docker containers (cli and compose), but yours is too complicated for me.
I really appreciate your effort for helping me, but I have to give up. Sorry

Is there anybody who got benotes running with docker on a synology?

**Reply by philipp-koch** [reply]
@Sylvio82, I struggled, too, but managed to overcome the problems, as [documented here](https://github.com/fr0tt/benotes/issues/43#issuecomment-1562008810). Hope that helps.
